### PR TITLE
[PBIOS-467] Typeahead manager fix

### PR DESCRIPTION
**Pull request by isismsilva**
**What does this PR do?** A clear and concise description with your runway ticket url.
Fix Typeahead when called inside a Dialog.

Set a unique handler for each typeahead.

**Screenshots:** Screenshots to visualize your addition/change

**How to test?** Steps to confirm the desired behavior:
4. See addition/change

### Checklist
- [ ] **LABELS** - Add a label: `breaking`, `bug`, `improvement`, `documentation`, or `enhancement`. See [Labels](https://github.com/powerhome/playbook-apple/labels) for descriptions.
- [ ] **RELEASES** - Add the appropriate label: `Ready for Testing` / `Ready for Release`
- [ ] **TESTING** - Have you tested your story?

### CODE REVIEW

File: Sources/Playbook/Components/Popover/Popover+Position.swift [hunk 1]
```diff
@@ -53,22 +53,22 @@ public extension Position {
...
+//      }
+//      if popoverFrame.maxX > maxX {
+//        let difference = popoverFrame.maxX - maxX
+//        popoverFrame.origin.x -= difference
+//      }
+//      if popoverFrame.maxY > maxY {
+//        let difference = popoverFrame.maxY - maxY
+//        popoverFrame.origin.y -= difference
+//      }
+//      #endif
```
Comment (welingtonramospwr) [inline Sources/Playbook/Components/Popover/Popover+Position.swift:71]: Is it still needed?

File: Sources/Playbook/Components/Typeahead/TypeaheadCatalog.swift [hunk 2]
```diff
@@ -14,31 +14,39 @@ public struct TypeaheadCatalog: View {
...
+            PBDoc(title: "Dialog") { dialog }
+            #endif
         }
         .onTapGesture {
             isFocused1 = false
             isFocused2 = false
         }
-        .popoverHandler()
+        .popoverHandler(id: 1)
+        .popoverHandler(id: 2)
```
Comment (welingtonramospwr) [inline Sources/Playbook/Components/Typeahead/TypeaheadCatalog.swift:42]: Do we need two different handlers here ?

File: Sources/Playbook/Components/Typeahead/TypeaheadCatalog.swift [hunk 3]
```diff
@@ -81,3 +143,7 @@ extension TypeaheadCatalog {
     //    }
     //  }
 }
```
Comment (welingtonramospwr) [inline Sources/Playbook/Components/Typeahead/TypeaheadCatalog.swift:145]: 👀

**Review by welingtonramospwr — DISMISSED** [review]

File: Sources/Playbook/Components/Typeahead/TypeaheadCatalog.swift (hunk 2 above)
Comment (isismsilva) [reply Sources/Playbook/Components/Typeahead/TypeaheadCatalog.swift:42]: yes. We need to manage each popover separetly so we can present different popovers in different view hierarchies.

File: Sources/Playbook/Components/Typeahead/TypeaheadCatalog.swift (hunk 3 above)
Comment (isismsilva) [reply Sources/Playbook/Components/Typeahead/TypeaheadCatalog.swift:123]: removed

File: Sources/Playbook/Components/Popover/Popover+Position.swift (hunk 1 above)
Comment (isismsilva) [reply Sources/Playbook/Components/Popover/Popover+Position.swift:71]: yes. We will need this to fix the margins. It's a future improvement.

**Review by welingtonramospwr — APPROVED** [review]

**Review by nidaqg — APPROVED** [review]

**Pull request merged**